shell/platform/tizen/external_texture_surface_gl_tizen.cc

```diff
@@ -105,11 +127,57 @@ bool ExternalTextureSurfaceGL::PopulateTexture(
   PFNEGLCREATEIMAGEKHRPROC n_eglCreateImageKHR =
       reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(
           eglGetProcAddress("eglCreateImageKHR"));
-  const EGLint attribs[] = {EGL_IMAGE_PRESERVED_KHR, EGL_TRUE, EGL_NONE,
-                            EGL_NONE};
-  EGLImageKHR egl_src_image =
-      n_eglCreateImageKHR(eglGetCurrentDisplay(), EGL_NO_CONTEXT,
-                          EGL_NATIVE_SURFACE_TIZEN, tbm_surface, attribs);
+  EGLImageKHR egl_src_image = nullptr;
+
+  if (state_->gl_extention ==
+      ExternalTextureGLExtention_EGL_TIZEN_image_native_surface) {
+    const EGLint attribs[] = {EGL_IMAGE_PRESERVED_KHR, EGL_TRUE, EGL_NONE,
+                              EGL_NONE};
+    egl_src_image =
+        n_eglCreateImageKHR(eglGetCurrentDisplay(), EGL_NO_CONTEXT,
+                            EGL_NATIVE_SURFACE_TIZEN, tbm_surface, attribs);
+  } else if (state_->gl_extention ==
+             ExternalTextureGLExtention_EGL_EXT_image_dma_buf_import) {
+    {
+      EGLint attribs[50];
+      int atti = 0;
+      tbm_bo tbo = NULL;
+      int bo_idx, num_planes, i;
+      int plane_fd_ext[4] = {
+          EGL_DMA_BUF_PLANE0_FD_EXT, EGL_DMA_BUF_PLANE1_FD_EXT,
+          EGL_DMA_BUF_PLANE2_FD_EXT, EGL_DMA_BUF_PLANE3_FD_EXT};
+      int plane_offset_ext[4] = {
+          EGL_DMA_BUF_PLANE0_OFFSET_EXT, EGL_DMA_BUF_PLANE1_OFFSET_EXT,
+          EGL_DMA_BUF_PLANE2_OFFSET_EXT, EGL_DMA_BUF_PLANE3_OFFSET_EXT};
+      int plane_pitch_ext[4] = {
+          EGL_DMA_BUF_PLANE0_PITCH_EXT, EGL_DMA_BUF_PLANE1_PITCH_EXT,
+          EGL_DMA_BUF_PLANE2_PITCH_EXT, EGL_DMA_BUF_PLANE3_PITCH_EXT};
+
+      attribs[atti++] = EGL_WIDTH;
+      attribs[atti++] = info.width;
+      attribs[atti++] = EGL_HEIGHT;
+      attribs[atti++] = info.height;
+      attribs[atti++] = EGL_LINUX_DRM_FOURCC_EXT;
+      attribs[atti++] = info.format;
+
+      num_planes = tbm_surface_internal_get_num_planes(info.format);
+      for (i = 0; i < num_planes; i++) {
+        bo_idx = tbm_surface_internal_get_plane_bo_idx(tbm_surface, i);
+        tbo = tbm_surface_internal_get_bo(tbm_surface, bo_idx);
```

Comment:
```suggestion
      int num_planes = tbm_surface_internal_get_num_planes(info.format);
      for (int i = 0; i < num_planes; i++) {
        int bo_idx = tbm_surface_internal_get_plane_bo_idx(tbm_surface, i);
        tbm_bo tbo = tbm_surface_internal_get_bo(tbm_surface, bo_idx);
```
https://google.github.io/styleguide/cppguide.html#Local_Variables

Reply:
👍